rclcpp/include/rclcpp/parameter_map.hpp

```diff
@@ -35,11 +35,13 @@ using ParameterMap = std::unordered_map<std::string, std::vector<Parameter>>;
 
 /// Convert parameters from rcl_yaml_param_parser into C++ class instances.
 /// \param[in] c_params C structures containing parameters for multiple nodes.
+/// \param[in] node_fqn a Fully Qualified Name of node, default value is nullptr.
+///   If it's not nullptr, return the relative node parameters belonging to this node_fqn.
 /// \returns a map where the keys are fully qualified node names and values a list of parameters.
 /// \throws InvalidParametersException if the `rcl_params_t` is inconsistent or invalid.
 RCLCPP_PUBLIC
 ParameterMap
-parameter_map_from(const rcl_params_t * const c_params);
+parameter_map_from(const rcl_params_t * const c_params, const char * node_fqn = nullptr);
```

Comment:
this breaks ABI

Reply:
I thought this would be acceptable since we do not break the user space? but technically, it breaks ABI. so after all, we cannot backport the fix to humble or galactic...

Reply:
You could backport if you instead add

```cpp
ParameterMap
parameter_map_from(const rcl_params_t * const c_params);

ParameterMap
parameter_map_from(const rcl_params_t * const c_params, const char * node_fqn);
```

and in the .cpp you can have a `static` common implementation

Reply:
Sorry, I didn't notice the ABI-breaking before.
Thanks for your suggestion, but it seems unnecessary to add a `static` function.
If I were wrong, I could update it again.

Reply:
> Thanks for your suggestion, but it seems unnecessary to add a static function.
If I were wrong, I would update it again.

Yes, sorry.
The current code seems not to break ABI.